medusa/common.py

```diff
@@ -601,6 +601,46 @@ def should_search(status, show_obj):
         return True
 
     @staticmethod
+    def should_replace(ep_status, old_quality, new_quality, allowed_qualities, preferred_qualities):
```

Comment:
Only remark I could have, is that "old_quality, new_quality, allowed_qualities, preferred_qualities" are all similar parameters. They could be easily passed as kwargs.